python/lsst/ctrl/mpexec/cmdLineFwk.py

```diff
@@ -199,7 +199,13 @@ def __init__(self, registry: Registry, args: argparse.Namespace, writeable: bool
             self.outputRun = None
         else:
             raise ValueError("Cannot write without at least one of (--output, --output-run).")
-        self.inputs = tuple(CollectionSearch.fromExpression(args.input)) if args.input else ()
+        # Recursively flatten any input CHAINED collections.  We do this up
+        # front so we can tell if the user passes the same inputs on subsequent
+        # calls, even though we also flatten when we define the output CHAINED
+        # collection.
+        self.inputs = CollectionSearch.fromExpression(
+            list(registry.queryCollections(args.input, flattenChains=True)) if args.input else ()
```

Comment:
`fromExpression` accepts iterables so `list()` should not be needed here, or if you want to make a list then `else` should also produce a list, not tuple?